sdk/go/common/workspace/plugins_test.go

```diff
@@ -541,6 +541,32 @@ func TestPluginDownload(t *testing.T) {
 			assert.Equal(t, expectedBytes, readBytes)
 		})
 	})
+	t.Run("Test Downloading From GitLab Releases", func(t *testing.T) {
+		t.Setenv("GITLAB_TOKEN", token)
+		version := semver.MustParse("1.23.4")
+		spec := PluginSpec{
+			PluginDownloadURL: "gitlab://gitlab.com/278964",
+			Name:              "mock-gitlab",
+			Version:           &version,
+			Kind:              PluginKind("resource"),
+		}
+		source, err := spec.GetSource()
+		assert.NoError(t, err)
+		getHTTPResponse := func(req *http.Request) (io.ReadCloser, int64, error) {
+			assert.Equal(t,
+				"https://gitlab.com/api/v4/projects/278964/releases/v1.23.4/downloads/"+
+					"pulumi-resource-mock-gitlab-v1.23.4-windows-arm64.tar.gz", req.URL.String())
+			assert.Equal(t, fmt.Sprintf("Bearer %s", token), req.Header.Get("Authorization"))
+			assert.Equal(t, "application/octet-stream", req.Header.Get("Accept"))
+			return newMockReadCloser(expectedBytes)
```

Comment:
So this exists from before, but we didn't need a custom mock readcloser here.
We can wrap a `bytes.Reader` or `strings.Reader` in [io.NopCloser](https://pkg.go.dev/io#NopCloser).

```go
return io.NopCloser(bytes.NewReader(expectedBytes))
```

Reply:
I've left the newMockReadCloser functions in because they're also returning length and a nil error, but I've killed the mockReaderCloser type and made those function just use io.NopCloser instead.